zipline/data/hdf5_daily_bars.py

```diff
@@ -726,21 +748,17 @@ def countries(self):
         return viewkeys(self._readers)
 
     def _country_code_for_assets(self, assets):
-        try:
-            country_codes = self._country_map.loc[assets]
-        except KeyError as exc:
-            raise_from(
-                NoDataForSid(
-                    'Assets not contained in daily pricing file: {}'.format(
-                        set(assets) - set(self._country_map)
-                    )
-                ),
-                exc
-            )
+        country_codes = self._country_map.get(assets)
 
-        unique_country_codes = country_codes.unique()
+        if country_codes is not None:
+            unique_country_codes = country_codes.dropna().unique()
+            num_countries = len(unique_country_codes)
+        else:
+            num_countries = 0
 
-        if len(unique_country_codes) > 1:
+        if num_countries == 0:
+            return self._country_map.iloc[0]
```

Comment:
Hmm...it feels a little dangerous to return a value here. As a consumer, I'm not sure I'd expect to get a random country back if no assets were found. Another slight worry I have is that this would break in a surprising way if we don't have any readers (we'd KeyError on the .iloc[0] call). 

One alternative would be to raise an error or return a sentinel value here and have the caller decide what to do in this case. For `load_raw_arrays`, for example, we could return an array containing all missing values without ever delegating to any underlying readers (which would work even if we don't have any such readers). I'm 50/50 on whether that compositionality improvement is with the extra complexity in call-sites though. 

Another option would be to just make this an error case, since most likely it's a bug if we hit this path in real-world code.

 Thoughts?

Reply:
Yeah, my thinking here was that we just want to route this request to a reader that can give us back an empty array. But these are good points, I think I'll just have this raise a `ValueError`.